Modules/main.c

```diff
@@ -101,6 +101,7 @@ pymain_err_print(int *exitcode_p)
     int exitcode;
     if (_Py_HandleSystemExit(&exitcode)) {
         *exitcode_p = exitcode;
+        Py_Exit(exitcode);
```

Comment:
This is needed in order to match behavior of `PyRUn_AnyFileExFlags`(which the old REPL called an still calls if `PYTHON_BASIC_REPL=1`) which somewhere calls `handle_system_exit` (`Python/pythonrun.c::L632)`) which handles `SystemExit` and calls `Py_Exit`.
That's why current `_pyrepl` doesn't work with exit codes as expected.

Reply:
In that case this path doesn't return as this calls `PyFinalize_Ex` and then `exit()`

Reply:
As well as previous approach with `PyRun_AnyFileExFlags`, so, is this a problem?

Reply:
Not sure I follow what you mean. Could you clarify ?

Reply:
Sorry for not being clear!
I mean, the previous approach with `PyRun_AnyFileExFlags` contains path with `handle_system_exit` which is also didn't return as it calls `PyFinalizeEx` and then `exit()`.